app/process/watcher/prometheus/watcher.go

```diff
@@ -73,37 +104,259 @@ func (pw Watcher) Watch(q qi.Queue) {
+				}
+			} else { // EVM
+				evm := pw.EVMClients[chainId].GetClient()
+				address := common.HexToAddress(pw.configuration.Bridge.EVMs[chainId].RouterContractAddress)
+
+				wrappedInstance, e := wtoken.NewWtoken(common.HexToAddress(asset), evm)
+				if e != nil {
+					panic(e)
+				}
+
+				routerInstance, e := router.NewRouter(address, evm)
+				if e != nil {
+					panic(e)
+				}
+
+				pw.setEvmAssetSupplyMetric(wrappedInstance, asset)
+				pw.setEvmAssetBalanceMetric(wrappedInstance, asset, address)
+				pw.setEvmCountMetric(routerInstance, asset)
+			}
+		}
+	}
+}
+
+func (pw Watcher) setBridgeAccAssetsMetrics(bridgeAcc *model.AccountsResponse) {
+	for _, token := range bridgeAcc.Balance.Tokens {
+		metric := pw.prometheusService.GetGauge(pw.bridgeAccAssetsMetrics[token.TokenID])
+		if metric == nil {
+			log.Infof("Skip metrics for Bridge Account asset with ID [%s]", token.TokenID)
+		} else {
+			metric.Set(float64(token.Balance))
+			log.Infof("The Bridge Account asset with ID [%s] has Balance = %f", token.TokenID, float64(token.Balance))
+		}
+	}
+}
+
+func (pw Watcher) setHederaAssetSupplyMetric(asset string, token *model.TokenResponse) {
+	totalSupplyMetric := pw.prometheusService.GetGauge(pw.supplyAssetsMetrics[asset])
+	pw.setHederaAssetMetric(totalSupplyMetric, token.TotalSupply, asset)
+}
+
+func (pw Watcher) setHederaNetworkSupply(asset string) {
+	supply, e := pw.mirrorNode.GetNetworkSupply()
+	if e != nil {
+		panic(e)
+	}
+	totalSupplyMetric := pw.prometheusService.GetGauge(pw.supplyAssetsMetrics[asset])
+	pw.setHederaAssetMetric(totalSupplyMetric, supply.TotalSupply, asset)
+}
+
+func (pw Watcher) setHederaAssetMetric(metric prometheus.Gauge, value string, asset string) {
+	parseValue, e := strconv.ParseFloat(value, 64)
+	if e != nil {
+		panic(e)
+	}
+	metric.Set(parseValue)
+	log.Infof("The Asset with ID [%s] has Total Supply = %f", asset, parseValue)
+}
+
+func (pw Watcher) setEvmAssetSupplyMetric(wrappedInstance *wtoken.Wtoken, asset string) {
+	totalSupply, e := wrappedInstance.TotalSupply(&bind.CallOpts{})
+	if e != nil {
+		panic(e)
+	}
+	totalSupplyMetric := pw.prometheusService.GetGauge(pw.supplyAssetsMetrics[asset])
+	pw.setEvmAssetMetric(totalSupplyMetric, asset, "Total Supply", totalSupply)
+}
+
+func (pw Watcher) setEvmAssetBalanceMetric(wrappedInstance *wtoken.Wtoken, asset string, address common.Address) {
+	balance, e := wrappedInstance.BalanceOf(&bind.CallOpts{}, address)
+	if e != nil {
+		panic(e)
+	}
+	balanceMetric := pw.prometheusService.GetGauge(pw.balanceAssetsMetrics[asset])
+	pw.setEvmAssetMetric(balanceMetric, asset, "Balance of", balance)
+}
+
+func (pw Watcher) setEvmCountMetric(routerInstance *router.Router, asset string) {
+	countMetric := pw.prometheusService.GetGauge(pw.countAssetsMetrics[asset])
+	count, e := routerInstance.NativeTokensCount(&bind.CallOpts{})
+	if e != nil {
+		panic(e)
+	}
+	pw.setEvmAssetMetric(countMetric, asset, "Count of", count)
+}
```

Comment:
I think it would be better to clarify what we are counting -> native tokens, in the logs and in the metrics

Reply:
In the docs as well

Reply:
fixed in https://github.com/LimeChain/hashport-validator/pull/381